vmdb/spec/models/service_template_provision_request_spec.rb

```diff
@@ -99,7 +101,7 @@
 
     it "finished with errors state" do
       @task_1_1.update_and_notify_parent({:state => "finished", :status => "Error", :message => "Error Message"})
-      @task_2_1.update_and_notify_parent({:state => "finished", :status => "Ok", :message => "Test Message"})
+      @task_2_1.update_and_notify_parent(:state => "finished", :status => "Error", :message => "Test Message")
```

Comment:
@mkanoor 
The "finished with errors state" test appears odd because the factory setup for this test isn't ideal.
Normally, we would create a service structure like this:
request -> 
        task_0  -> 
                   -> task_1
                          -> task_1_1
                          -> task_1_2
                    -> task_2
                          -> task_2_1
This test setup is:
request -> 
                   -> task_1
                          -> task_1_1
                   -> task_2
                          -> task_2_1
I changed the test criteria because the task 1_1 error state causes the task_1 state to be set to finished, and the same is true for task 2_1 which will cause the task_2 state to be finished. Only when all of the tasks are finished, does the request state get set to finished.